lib/index.js

```diff
@@ -89,8 +89,12 @@ module.exports = function tinyreq (options, callback) {
             opt_callback(e, null, res)
         }).on("end", () => {
             str.emit("end")
-            body = options.encoding === null || options.encoding === "buffer" ? Buffer.concat(body, bodyLength) : body.join("")
-            opt_callback(null, body, res)
+            if (res.headers["content-encoding"] === "gzip" && options.encoding === "utf8") {
+                // TODO gunzip
```

Comment:
@coltaemanuela Use `zlib` to gunzip the `body` here. Here's a useful snippet: https://stackoverflow.com/a/19035763/1420197